### ISSUE TITLE
cannot pass METEOR_SETTINGS to meteor run

### ISSUE THREAD

**Issue by macrozone** [meteor/meteor#9907]
Because of https://github.com/meteor/meteor/blob/ac085d96fb79dfc03d28af8d4a3de50e99313ce4/tools/runners/run-app.js#L177

its not possible to pass `METEOR_SETTINGS` to `meteor` command. This prevents to pass METEOR_SETTINGS from something else but a file. E.g. we generate it from a secret password store.

Is it possible to bypass that?

My suggestion is to remove that else-case as it feels arbitrary for me. If someone would pass `METEOR_SETTINGS`-env-var to meteor, they usually know what to they do.

**Comment by macrozone** [comment]
i also tried something like `mycommand | meteor --settings  -`, but this does not work either :-((

**Comment by larswolter** [comment]
Passing meteor settings thru the environment is supported when running as a bundle. Supplying them in settings.json is just for development where you can easily handle multiple configurations using files. 

https://docs.meteor.com/api/core.html#Meteor-settings

Normally during development the settings should not contain any secret stuff. We would have problems using the environment during development, because multiple instances are usually run during development.

**Comment by macrozone** [comment]
@larswolter I understand how `--settings settings.json` works and used that before. Also passing METEOR_SETTINGS via env vars seems to work perfectly, but its just actively prevented because of... well... "educational" reasons. 🤔

local development may require secret env vars, e.g. api keys to make request against other services. using env vars allows to use tools like https://direnv.net and with some scripting magic you can even pull env vars out of password store tools like `pass` or similar. Creating a json file "on-the-fly" is much harder.

**Comment by macrozone** [comment]
no, bad stalebot!

there is a PR open, hope it gets merged

**Comment by filipenevola** [comment]
thanks @macrozone, I already labeled as `confirmed` then stale bot will ignore it.

**Comment by filipenevola** [comment]
This fix is going to be released on Meteor 2.4. You can already test it on latest RC #11488

**Comment by macrozone** [comment]
The linked PR does not solve the issue. It simply adds an additional check for `env.NODE_ENV === 'development'`, which is true anyway in most cases as you will usually use the `meteor` command in development.

Goal is that you can use METEOR_SETTINGS env var as an alternative to `--settings`. 

@filipenevola should i create a new issue?

**Comment by StorytellerCZ** [comment]
@macrozone feel free to create a PR

**Comment by macrozone** [comment]
> @macrozone feel free to create a PR

i created one 2 years ago: https://github.com/meteor/meteor/pull/10605

(CLA did not work back then, so it got closed without providing any help.)

It puzzles me that there is a discussion about this particular issue at all. There is nearly no value in forcing users to use `--settings` (yes, it refreshes automatically if you change settings.yaml, hooray), but thats not reason enough to purpously break the alternative of using an env var. I explained why this is important for some use cases in great detail here and in the pull request.
